specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-08-01-preview/settings.json

```diff
@@ -0,0 +1,333 @@
+            "readOnly": true
+          },
+          "message": {
+            "description": "Error message indicating why the operation failed.",
+            "type": "string",
+            "readOnly": true
+          }
+        }
+      },
+      "SettingProxyResource": {
+        "description": "The Resource model definition.",
+        "properties": {
+          "id": {
+            "readOnly": true,
+            "type": "string",
+            "description": "Resource Id."
+          },
+          "name": {
+            "readOnly": true,
+            "type": "string",
+            "description": "Resource name."
+          },
+          "type": {
+            "readOnly": true,
+            "type": "string",
+            "description": "Resource type."
+          }
+        },
+        "type": "object",
+        "x-ms-azure-resource": true
+      },
+      "Setting": {
+        "description": "Setting definition.",
+        "type": "object",
+        "allOf": [
+          {
+            "$ref": "#/definitions/SettingProxyResource"
+          }
+        ],
+        "properties": {
+          "properties": {
+            "x-ms-client-flatten": true,
+            "$ref": "#/definitions/TagInheritanceProperties",
+            "title": "Tag Inheritance properties"
+          }
+        }
+      },
+      "SettingsListResult": {
+        "description": "Setting list result. It contains a list of settings.",
+        "type": "object",
+        "properties": {
+          "value": {
+            "description": "The list of settings.",
+            "type": "array",
+            "readOnly": true,
+            "items": {
+              "$ref": "#/definitions/Setting"
+            }
+          }
+        }
+      },
+      "TagInheritanceProperties": {
+        "type": "object",
+        "description": "The properties of the tag inheritance setting.",
+        "required": [
+          "preferContainerTags"
+        ],
+        "properties": {
+          "preferContainerTags": {
+            "description": "Prefer Container tags to override container tags with resource tags in case of conflicts.",
+            "type": "boolean"
+          }
+        }
+      }
+    },
+    "parameters": {
+      "apiVersionParameter": {
+        "name": "api-version",
+        "in": "query",
+        "required": true,
+        "type": "string",
+        "description": "Version of the API to be used with the client request."
+      },
+      "scopeParameter": {
+        "name": "scope",
+        "in": "path",
+        "required": true,
+        "type": "string",
+        "description": "The scope associated with setting. This includes 'subscriptions/{subscriptionId}' for subscription scope, 'providers/Microsoft.Billing/billingAccounts/{billingAccountId}' for Billing Account scope, 'providers/Microsoft.Billing/billingAccounts/{billingAccountId}/billingProfiles/{billingProfileId}' for billing profile scope.",
+        "x-ms-parameter-location": "method"
+      },
```

Comment:
Can you just have a single definition of scopeParameter in your common-types.json referenced from all your other json files?